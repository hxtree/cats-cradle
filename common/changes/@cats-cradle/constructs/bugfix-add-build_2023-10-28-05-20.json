{
  "changes": [
    {
      "packageName": "@cats-cradle/constructs",
      "comment": "",
      "type": "none"
    }
  ],
  "packageName": "@cats-cradle/constructs"
}