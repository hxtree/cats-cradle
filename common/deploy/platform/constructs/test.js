'use strict';
(() => {
  // src/__tests__/test.ts
  console.log('test');
})();
