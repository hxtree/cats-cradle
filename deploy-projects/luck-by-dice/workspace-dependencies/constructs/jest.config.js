module.exports = {
  testEnvironment: 'node',
  roots: ['<rootDir>/src/__tests__'],
  testMatch: ['**/*.test.ts'],
  transform: {
    '^.+\\.tsx?$': 'ts-jest',
  },
};
