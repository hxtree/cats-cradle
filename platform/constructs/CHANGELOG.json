{
  "name": "@cats-cradle/constructs",
  "entries": [
    {
      "version": "0.0.13",
      "tag": "@cats-cradle/constructs_v0.0.13",
      "date": "Tue, 05 Mar 2024 05:29:22 GMT",
      "comments": {
        "patch": [
          {
            "comment": "update deps"
          }
        ],
        "dependency": [
          {
            "comment": "Updating dependency \"@cats-cradle/base-nodejs\" to `1.0.12`"
          }
        ]
      }
    },
    {
      "version": "0.0.12",
      "tag": "@cats-cradle/constructs_v0.0.12",
      "date": "Sun, 04 Feb 2024 05:29:40 GMT",
      "comments": {
        "dependency": [
          {
            "comment": "Updating dependency \"@cats-cradle/base-nodejs\" to `1.0.11`"
          }
        ]
      }
    },
    {
      "version": "0.0.11",
      "tag": "@cats-cradle/constructs_v0.0.11",
      "date": "Sat, 03 Feb 2024 08:29:12 GMT",
      "comments": {
        "patch": [
          {
            "comment": "api gateway construct update"
          }
        ]
      }
    },
    {
      "version": "0.0.10",
      "tag": "@cats-cradle/constructs_v0.0.10",
      "date": "Fri, 02 Feb 2024 05:50:03 GMT",
      "comments": {
        "patch": [
          {
            "comment": "add dns api endpoint"
          }
        ]
      }
    },
    {
      "version": "0.0.9",
      "tag": "@cats-cradle/constructs_v0.0.9",
      "date": "Thu, 28 Dec 2023 16:54:41 GMT",
      "comments": {
        "patch": [
          {
            "comment": "upgrade typescript 5.3.3"
          }
        ],
        "dependency": [
          {
            "comment": "Updating dependency \"@cats-cradle/base-nodejs\" to `1.0.10`"
          }
        ]
      }
    },
    {
      "version": "0.0.8",
      "tag": "@cats-cradle/constructs_v0.0.8",
      "date": "Tue, 12 Dec 2023 03:15:41 GMT",
      "comments": {
        "dependency": [
          {
            "comment": "Updating dependency \"@cats-cradle/base-nodejs\" to `1.0.9`"
          }
        ]
      }
    },
    {
      "version": "0.0.7",
      "tag": "@cats-cradle/constructs_v0.0.7",
      "date": "Mon, 13 Nov 2023 02:11:53 GMT",
      "comments": {
        "patch": [
          {
            "comment": "update packages"
          }
        ]
      }
    },
    {
      "version": "0.0.6",
      "tag": "@cats-cradle/constructs_v0.0.6",
      "date": "Sun, 05 Nov 2023 18:41:09 GMT",
      "comments": {
        "dependency": [
          {
            "comment": "Updating dependency \"@cats-cradle/base-nodejs\" to `1.0.8`"
          }
        ]
      }
    },
    {
      "version": "0.0.5",
      "tag": "@cats-cradle/constructs_v0.0.5",
      "date": "Sat, 04 Nov 2023 19:40:56 GMT",
      "comments": {
        "patch": [
          {
            "comment": "unify unit tests"
          }
        ],
        "dependency": [
          {
            "comment": "Updating dependency \"@cats-cradle/base-nodejs\" to `1.0.7`"
          }
        ]
      }
    },
    {
      "version": "0.0.4",
      "tag": "@cats-cradle/constructs_v0.0.4",
      "date": "Sat, 14 Oct 2023 22:49:51 GMT",
      "comments": {
        "patch": [
          {
            "comment": "add layers prop"
          }
        ]
      }
    },
    {
      "version": "0.0.3",
      "tag": "@cats-cradle/constructs_v0.0.3",
      "date": "Wed, 04 Oct 2023 03:44:49 GMT",
      "comments": {
        "patch": [
          {
            "comment": "bump packages"
          }
        ]
      }
    },
    {
      "version": "0.0.2",
      "tag": "@cats-cradle/constructs_v0.0.2",
      "date": "Tue, 27 Jun 2023 18:44:40 GMT",
      "comments": {
        "patch": [
          {
            "comment": "upgrade to node 18 lts"
          }
        ]
      }
    }
  ]
}
