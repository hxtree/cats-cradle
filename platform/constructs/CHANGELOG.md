# Change Log - @cats-cradle/constructs

This log was last generated on Sun, 04 Feb 2024 05:29:40 GMT and should not be manually modified.

## 0.0.12
Sun, 04 Feb 2024 05:29:40 GMT

_Version update only_

## 0.0.11
Sat, 03 Feb 2024 08:29:12 GMT

### Patches

- api gateway construct update

## 0.0.10
Fri, 02 Feb 2024 05:50:03 GMT

### Patches

- add dns api endpoint

## 0.0.9
Thu, 28 Dec 2023 16:54:41 GMT

### Patches

- upgrade typescript 5.3.3

## 0.0.8
Tue, 12 Dec 2023 03:15:41 GMT

_Version update only_

## 0.0.7
Mon, 13 Nov 2023 02:11:53 GMT

### Patches

- update packages

## 0.0.6
Sun, 05 Nov 2023 18:41:09 GMT

_Version update only_

## 0.0.5
Sat, 04 Nov 2023 19:40:56 GMT

### Patches

- unify unit tests

## 0.0.4
Sat, 14 Oct 2023 22:49:51 GMT

### Patches

- add layers prop

## 0.0.3
Wed, 04 Oct 2023 03:44:49 GMT

### Patches

- bump packages

## 0.0.2
Tue, 27 Jun 2023 18:44:40 GMT

### Patches

- upgrade to node 18 lts

