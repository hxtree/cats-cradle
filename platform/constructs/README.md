L1 constructs
L2 constructs

https://docs.aws.amazon.com/cdk/v2/guide/constructs.html