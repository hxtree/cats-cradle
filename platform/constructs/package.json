{
  "name": "@cats-cradle/constructs",
  "version": "0.0.1",
  "module": "commonjs",
  "main": "./dist/src/index.js",
  "scripts": {
    "build": "tsc",
    "watch": "tsc -w",
    "test": "jest",
    "lint": "eslint --format visualstudio \"./{stacks,src}/*.ts\"  --fix",
    "lint:ci": "eslint --format visualstudio \"./{stacks,src}/*.ts\"  --fix-dry-run",
    "depcheck": "npx depcheck"
  },
  "dependencies": {
    "aws-lambda": "1.0.7",
    "class-transformer": "0.5.1",
    "class-validator": "0.13.2",
    "@nestjs/common": "9.4.0",
    "@nestjs/core": "9.2.1",
    "@nestjs/terminus": "9.1.1",
    "@nestjs/platform-express": "9.3.9",
    "aws-serverless-express": "~3.4.0",
    "@vendia/serverless-express": "4.10.1",
    "express": "4.18.2",
    "node-cache": "5.1.2",
    "reflect-metadata": "0.1.13",
    "rxjs": "7.5.6",
    "mongoose": "~6.9.1",
    "source-map-support": "~0.5.21",
    "@nestjs/swagger": "~6.1.3",
    "uuid": "~9.0.0",
    "lodash": "~4.17.21",
    "@types/lodash": "~4.14.191",
    "@cats-cradle/bundlers": "workspace:*"
  },
  "devDependencies": {
    "aws-cdk-lib": "2.47.0",
    "constructs": "^10.1.303",
    "@types/jest": "28.1.8",
    "@types/node": "^16.0.0",
    "@types/aws-lambda": "~8.10.114",
    "jest": "29.5.0",
    "ts-jest": "29.1.0",
    "typescript": "5.0.2",
    "ts-node": "10.9.1",
    "esbuild": "~0.17.5",
    "fs": "~0.0.1-security",
    "path": "~0.12.7",
    "@types/express": "~4.17.14",
    "@types/uuid": "~8.3.4",
    "eslint": "8.38.0",
    "@cats-cradle/eslint-config": "1.0.5"
  },
  "exports": {
    ".": "./dist/src/index.js",
    "./api-endpoint": "./dist/src/api-endpoint/index.js",
    "./aws-account": "./dist/src/aws-accounts/index.js",
    "./microservice": "./dist/src/microservice/index.js",
    "./nestjs": "./dist/src/nestjs/index.js"
  }
}
