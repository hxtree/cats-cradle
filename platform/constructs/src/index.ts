export * from './microservice/microservice';
export * from './api-endpoint/api-endpoint';
export * from './api-endpoint/get-base-url';
export * from './aws-accounts';
export * from './nestjs/nestjs';
