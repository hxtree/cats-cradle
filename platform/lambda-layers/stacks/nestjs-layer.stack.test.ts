import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { Construct } from 'constructs';
import { NestJsLayerStack } from './nestjs-layer.stack';

describe('NestJsLambdaLayerStack', () => {
  it('should match snapshot test', () => {
    class TestStack extends cdk.Stack {
      constructor(scope: Construct, id: string, props?: cdk.StackProps) {
        super(scope, id, props);

        new NestJsLayerStack(this, 'test-stack');
      }
    }

    const app = new cdk.App();
    const stack = new TestStack(app, 'TestStack');
    const template = Template.fromStack(stack);

    // TODO consider reworking to use DNS instead of incrementing and deploying each time
    const customSnapshotSerializer = {
      test: (val: any) => typeof val === 'string',
      print: (val: any) => val
        .replace(/deployment\d{8}T\d{9}Z[^:]+/, 'deployment_UNIQUE_HASH')
        .replace(/[a-fA-F0-9]+\.zip/, 'UNIQUE_HASH.zip'),
    };
    expect.addSnapshotSerializer(customSnapshotSerializer);

    expect(template.toJSON()).toMatchSnapshot();
  });
});
