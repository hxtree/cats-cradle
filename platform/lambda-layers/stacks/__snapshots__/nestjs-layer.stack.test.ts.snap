// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NestJsLambdaLayerStack should match snapshot test 1`] = `
{
  Parameters: {
    BootstrapVersion: {
      Default: /cdk-bootstrap/hnb659fds/version,
      Description: Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip],
      Type: AWS::SSM::Parameter::Value<String>,
    },
  },
  Resources: {
    teststackNestedStackteststackNestedStackResourceFBD69DE9: {
      DeletionPolicy: Delete,
      Properties: {
        TemplateURL: {
          Fn::Join: [
            ,
            [
              https://s3.,
              {
                Ref: AWS::Region,
              },
              .,
              {
                Ref: AWS::URLSuffix,
              },
              /,
              {
                Fn::Sub: cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region},
              },
              /535250180b162a702a8bbbaf274591b734e1b1093c2d3d538f1768b27b1ce1e4.json,
            ],
          ],
        },
      },
      Type: AWS::CloudFormation::Stack,
      UpdateReplacePolicy: Delete,
    },
  },
  Rules: {
    CheckBootstrapVersion: {
      Assertions: [
        {
          Assert: {
            Fn::Not: [
              {
                Fn::Contains: [
                  [
                    1,
                    2,
                    3,
                    4,
                    5,
                  ],
                  {
                    Ref: BootstrapVersion,
                  },
                ],
              },
            ],
          },
          AssertDescription: CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.,
        },
      ],
    },
  },
}
`;
